libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt

```diff
@@ -36,55 +63,66 @@ object MapboxMetricsReporter : MetricsReporter {
         accessToken: String,
         userAgent: String
     ) {
-        mapboxTelemetry = MapboxTelemetry(context, accessToken, userAgent)
-        mapboxTelemetry.enable()
-    }
-
-    // For test purposes only
-    internal fun init(
-        mapboxTelemetry: MapboxTelemetry,
-        jobControlFactory: InternalJobControlFactory,
-    ) {
-        this.mapboxTelemetry = mapboxTelemetry
-        ioJobController = jobControlFactory.createIOScopeJobControl()
-        mapboxTelemetry.enable()
+        val eventsServerOptions = EventsServerOptions(accessToken, userAgent, null)
+        eventsService = EventsServiceProvider.provideEventsService(eventsServerOptions)
+        telemetryService = TelemetryServiceProvider.provideTelemetryService(eventsServerOptions)
+        eventsService.registerObserver(eventsServiceObserver)
     }
 
     /**
      * Toggle whether or not you'd like to log [mapboxTelemetry] events.
      *
      * @param isDebugLoggingEnabled true to enable logging, false to disable logging
      */
+    @Deprecated("toggleLogging is deprecated and do no operations")
     @JvmStatic
     fun toggleLogging(isDebugLoggingEnabled: Boolean) {
-        mapboxTelemetry.updateDebugLoggingEnabled(isDebugLoggingEnabled)
+        // do nothing
     }
 
     /**
-     * Disables metrics reporting and ends [mapboxTelemetry] session.
+     * Disables metrics reporting and ends [EventsServiceInterface] and [TelemetryService] session.
```

Comment:
Current implementation does not stop `EventsService` and `TelemetryService`.

What should this method do? If it must stop interaction and location events sending (for the entire application and all Mapbox SDKs we have in the application) we need to `setEventsCollectionState(false)` here. 
If it should not stop events collection we need to remove `Disables metrics reporting and ends [EventsServiceInterface] and [TelemetryService] session.` from method description.

Reply:
> we need to setEventsCollectionState(false) here.

@tatiana-yan then it stops collecting metrics from every single SDK (nav, maps, search), right? It's not fit for the current method, let's change the description

Reply:
Actually, we're not recommending using this API at all. To align the API I'll add internal flags 👍 

>And is there some use-case when we need to stop collecting navigation events but not maps/search/locations?  

don't think this must be done via MapboxReporter API

Reply:
Capturing from a chat, @RingerJK to cut a ticket to discuss our recommendations on how developers/end-users should opt-out of telemetry collection (since `TelemetryUtils#setEventsCollectionState` is documented as `internal`). Not a blocker for this PR but has to be resolved before the changes go stable.

Reply:
ref `CORESDK-1357`